File: v_javascript/test_admin_login.js
var webdriver = require('selenium-webdriver'),
    chrome = require('selenium-webdriver/chrome'),
    // firefox = require('selenium-webdriver/firefox'),
    // safari = require('selenium-webdriver/safari'),
    test = require('selenium-webdriver/testing'),
    helper = require('./helpers.js');

test.describe('LiteCart Admin Login', function() {
    var driver;

    test.before(function() {
        var options = new chrome.Options();
        options.addArguments(["start-fullscreen"]);
        // var ff_options = new firefox.Options().setBinary('/Applications/FirefoxNightly.app/Contents/MacOS/firefox');

        driver = new webdriver.Builder()
            .forBrowser('chrome')
            .setChromeOptions(options)
            // .forBrowser('firefox')
            // .setFirefoxOptions(ff_options)
            // .forBrowser('safari')
            .build();
        driver.getCapabilities().then(function(caps) {
            console.log(caps);
        });
    });

    test.it('should login with valid credentials', function() {
        helper.navigateToAdminPortal(driver);
        helper.adminLogin(driver);
        helper.waitForElementToBeVisible(driver, '#box-apps-menu-wrapper');
    });

    test.after(function() {
        driver.quit();
    });
});

File: v_python/test/test_admin_login.py
# import pytest
# from selenium import webdriver
# from helpers import admin_login
# from helpers import navigate_to_admin_portal
# from helpers import wait_for_element_to_be_visible
#
#
# @pytest.fixture
# def driver(request):
#     wd = webdriver.Chrome()
#     # wd = webdriver.Firefox()
#     # wd = webdriver.Safari()
#     # wd = webdriver.Firefox(capabilities={'marionette': False},
#     #                        firefox_binary='/Applications/Firefoxx.app/Contents/MacOS/firefox-bin')
#     # wd = webdriver.Firefox(firefox_binary='/Applications/FirefoxNightly.app/Contents/MacOS/firefox')
#     print(wd.capabilities)
#     request.addfinalizer(wd.quit)
#     return wd
#
#
# def test_valid_admin_login(driver):
#     navigate_to_admin_portal(driver)
#     admin_login(driver)
#     wait_for_element_to_be_visible(driver, '#box-apps-menu-wrapper')


def test_valid_admin_login(app):
    app.navigation.go_to_admin_root()
    app.session.admin_smart_login()
